content/doc/book/pipeline/pipeline-best-practices.adoc

```diff
@@ -0,0 +1,97 @@
+---
+layout: section
+title: Pipeline Best Practices
+---
+ifdef::backend-html5[]
+:notitle:
+:description:
+:author: Alex Taylor
+:email: jenkinsci-docs@googlegroups.com
+:sectanchors:
+ifdef::env-github[:imagesdir: ../resources]
+ifndef::env-github[:imagesdir: ../../resources]
+:hide-uri-scheme:
+:toc:
+endif::[]
+
+= Pipeline Best Practices
+
+== General
+
+=== Pipeline functions as Glue
+
+Make sure to use pipeline steps as a way of wiring together a set of actions rather than the main functionality. 
+This means that rather than relying on pipeline functionality(groovy or pipeline steps) to drive the build process forward, it is better to use single steps(such as `sh`) to accomplish multiple parts of the build. 
+The less pipeline functions used, the less memory will consumed up on master. 
```

Comment:
By itself, I don't think this claim makes sense. Running a Pipeline function does not in and of itself require more memory. Maybe rephrase to something like this:

```
As the complexity of Pipeline scripts increase, they require more resources (CPU, memory, storage) on the master.
```